src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -377,6 +480,54 @@ public void OnEventCommand(EventCommandEventArgs command)
                 }
             }
 
+            private void IncrementRefCount(string uniqueIdentifier, EventCommandEventArgs command)
+            {
+                // Could be unsafe if UniqueIdentifier protocol isn't followed
+                if (command.Arguments!.TryGetValue("UniqueIdentifier", out string? uniqueIdentifierArg))
+                {
+                    uniqueIdentifier = uniqueIdentifierArg!;
+                }
+
+                if (!_sharedSessionIds.Contains(uniqueIdentifier))
+                {
+                    _sharedSessionIds.Add(uniqueIdentifier);
+                    _sharedSessionRefCount += 1;
```

Comment:
Not my codebase but are you intentionally not doing interlocked on this case vs the others above